src/v2/components/UserSettings/index.tsx

```diff
@@ -0,0 +1,382 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Query, Mutation, MutationFn } from 'react-apollo'
+import { Form, Field } from 'react-final-form'
```

Comment:
Using final-form which is honestly pretty awesome.

Reply:
What's it give you? I generally don't get why these form libs exist tbh

Reply:
If you’ll notice, it handles all the complex state management. Not just the values but the errors, whether the form has been touched, etc. I don’t enjoy writing that stuff repeatedly when a lib can easily handle the standard patterns. And I don’t reckon I could write it better than someone else who is thinking specifically about forms. 

I originally added it because I wanted a nicer way to handle the custom home path which it handles easily.

Reply:
I just think dependencies for simple things like this add more complexity in the long run. Whatever makes you happy, ultimately.